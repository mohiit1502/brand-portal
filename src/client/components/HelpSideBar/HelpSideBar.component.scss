.c-HelpSideBar {
  border-right: 1px solid #CCCCCC;

  .c-HelpSideBar__header {
      padding-left: 0.5rem;
      padding-bottom: 0.6rem;
      margin-bottom: 0;
  }

  &.h-90 {
    height: 90%;
  }

  .c-HelpSideBar__categories {
    list-style: none;
    padding-left: unset;
  }

  .c-HelpSideBar__category {
    color: #8C8C8C;
    line-height: 1.2;
    padding: 0.95rem 0.5rem 0.95rem;
    // margin-bottom: 1.85rem;
    cursor: pointer;

    &.active {
        color: #000000;
        font-weight: bolder;
    }

    &:hover {
        background: #FAFAFA;
        font-weight: bolder;
    }
  }
}
