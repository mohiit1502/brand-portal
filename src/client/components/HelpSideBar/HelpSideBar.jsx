import React from "react";
import PropTypes from "prop-types";
import mixpanel from "../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../constants/MixPanelConsants";

import "./HelpSideBar.component.scss";

const HelpSideBar = ({activeTab, categoryHeader, categories, setActiveTab}) => {

  const onClickHandler = categoryKey => {
    setActiveTab(categoryKey);
    mixpanel.helpSectionEvents(MIXPANEL_CONSTANTS.HELP_CENTER_EVENTS.VIEW_HELP_MENU, {id: categoryKey});
  };

  const categoriesRendered = categories && Object.keys(categories).map(categoryKey => {
    const category = categories[categoryKey];
    return (<li key={categoryKey} className={`c-HelpSideBar__category${activeTab === categoryKey ? " active" : ""}`}
      onClick={() => {onClickHandler(categoryKey);}}>
      {category.categoryText}
      </li>);
  });

  return (
    <div className="c-HelpSideBar h-90">
      <div className="c-HelpSideBar__header h4">{categoryHeader}</div>
      <ul id="helpSideBar" className="c-HelpSideBar__categories">
        {categoriesRendered}
      </ul>
    </div>
  );
};

HelpSideBar.propTypes = {
  activeTab: PropTypes.string,
  categories: PropTypes.object,
  categoryHeader: PropTypes.string,
  setActiveTab: PropTypes.func
};

export default HelpSideBar;
