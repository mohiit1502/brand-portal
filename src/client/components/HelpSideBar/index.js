import HelpSideBar from './HelpSideBar.jsx';

export default HelpSideBar;