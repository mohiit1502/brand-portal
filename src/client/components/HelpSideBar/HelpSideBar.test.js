/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import HelpSideBar from "./HelpSideBar";

describe("HelpSideBar", () => {
   it("renders without error", () => {

   });
});
