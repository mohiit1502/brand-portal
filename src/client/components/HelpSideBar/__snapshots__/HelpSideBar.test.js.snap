// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpSideBar renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockNextContext
    pathname="/help"
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/help",
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/help",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <withRouter(HelpSideBar)
          activeTab=""
          categories={Object {}}
          categoryHeader=""
          history={Object {}}
          setActiveTab={[Function]}
        >
          <HelpSideBar
            activeTab=""
            categories={Object {}}
            categoryHeader=""
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/help",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/help",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            setActiveTab={[Function]}
          >
            <div
              className="c-HelpSideBar h-90"
            >
              <div
                className="c-HelpSideBar__header h4"
              />
              <ul
                className="c-HelpSideBar__categories"
                id="helpSideBar"
              />
            </div>
          </HelpSideBar>
        </withRouter(HelpSideBar)>
      </Router>
    </MemoryRouter>
  </MockNextContext>
</Provider>
`;
